daemon/execdriver/driver.go

```diff
@@ -130,18 +130,12 @@ type UTS struct {
 	HostUTS bool `json:"host_uts"`
 }
 
-// NetworkInterface contains all network configs for a driver
-type NetworkInterface struct {
-	Gateway              string `json:"gateway"`
-	IPAddress            string `json:"ip"`
-	IPPrefixLen          int    `json:"ip_prefix_len"`
-	MacAddress           string `json:"mac"`
-	Bridge               string `json:"bridge"`
-	GlobalIPv6Address    string `json:"global_ipv6"`
-	LinkLocalIPv6Address string `json:"link_local_ipv6"`
-	GlobalIPv6PrefixLen  int    `json:"global_ipv6_prefix_len"`
-	IPv6Gateway          string `json:"ipv6_gateway"`
-	HairpinMode          bool   `json:"hairpin_mode"`
+// CommonNetworkInterface contains the network configs for a driver that
+// are common across platforms.
+type CommonNetworkInterface struct {
```

Comment:
Well I would argue that this is a pattern we've already followed (and were actively advised to do). For example CommonConfig: https://github.com/docker/docker/blob/master/daemon/config.go#L16 and CommonContainer https://github.com/docker/docker/blob/master/daemon/container.go#L61. Certainly it doesn't make sense for example to have HairpinMode in Windows - that is highly unlikely to ever be supported, so splitting that out makes sense. It seems to also make sense to me to move other things back as and when the platform ends up supporting them. There's very little complex here in the code base IMO - in fact factoring this out hasn't changed any Linux source apart from the type declaration. The other argument for keeping it would be that I'm sure I'd get pushback if I added PortMappings in the unified structure which is only needed on Windows. I honestly think this is much cleaner :)